pretyping/evarsolve.ml

```diff
@@ -1486,8 +1486,8 @@ let solve_candidates unify flags env evd (evk,argsv) rhs =
           (* solve_candidates might have been called recursively in the mean *)
           (* time and the evar been solved by the filtering process *)
          if Evd.is_undefined evd evk then
-           let evd' = Evd.define evk c evd in
-             check_evar_instance unify flags env evd' evk c
+           let evd' = check_evar_instance unify flags env evd evk c in
+           Evd.define evk c evd'
```

Comment:
Shouldn't we have a single function which both check and assigns the evar?
I did some git grep and it seems doable, all uses of `check_evar_instance` are next to an `Evd.define`

Reply:
> Shouldn't we have a single function which both check and assigns the evar? I did some git grep and it seems doable, all uses of `check_evar_instance` are next to an `Evd.define`

Indeed, that would make a lot of sense.